### PR TITLE
Add Cluster foreign key to k8s objects

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

Adds cluster foreign key so we don't need to worry about pruning when clusters are deleted

@c-du We will need to be very careful on migration here

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

**Review by md2119 — CHANGES_REQUESTED** [review]
Cluster postgres store uses delete+copyFrom for batch operations. The delete will cause all the dependencies to be deleted. To avoid unintended cascade deletes from batch updates, add `no-fk-constraint` to proto or remove `copyFrom` from Cluster

**Comment by connorgorman** [comment]
> Cluster postgres store uses delete+copyFrom for batch operations. The delete will cause all the dependencies to be deleted. To avoid unintended cascade deletes from batch updates, add no-fk-constraint to proto or remove copyFrom from Cluster

@md2119 There are no batch operations for Clusters so this is actually safe

**Review by c-du — APPROVED** [review]

**Comment by md2119** [comment]
> > Cluster postgres store uses delete+copyFrom for batch operations. The delete will cause all the dependencies to be deleted. To avoid unintended cascade deletes from batch updates, add no-fk-constraint to proto or remove copyFrom from Cluster
> 
> @md2119 There are no batch operations for Clusters so this is actually safe, but do need to be careful about enabling that in the future

Let me rephrase the statement. This change relies on store layer batch functionality not being exposed. Furthermore, it requires the store's consumers to know the implementation details, which should not be the case. However, there is no check to prohibit that. I recommend we add a flag to generator to skip generating batch functionality.

**Review by md2119 — APPROVED** [review]
rebase and merge
